amarna/rules/post_process_rules/StorageVarCollisionRule.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict, List
+from amarna.Result import ResultMultiplePositions, result_multiple_positions
+
+from amarna.rules.gatherer_rules.DeclaredFunctionsGatherer import (
+    DeclaredFunctionsGatherer,
+    FunctionType,
+)
+
+
+class StorageVarCollisionRule:
+    """
+    Storage variable is declared multiple times across files.
+    """
+
+    RULE_TEXT = "[This](0) storage variable is also declared [here](1)."
+    RULE_NAME = "storage-var-collision"
+
+    def run_rule(self, gathered_data: Dict) -> List[ResultMultiplePositions]:
+        declared_functions: List[FunctionType] = gathered_data[
+            DeclaredFunctionsGatherer.GATHERER_NAME
+        ]
+
+        results = []
+        seen = {}
```

Comment:
Minor, but for information here you could use a [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict):
```python
from collections import defaultdict

...
seen = defaultdict(list)
```

This allows to skip the dict initialization (it's initialized with an empty list automatically). It would then allow you to remove the if/else initialization:
```python
            if "storage_var" in func.decorators:
                seen[func.name].append(func)
```

Reply:
it does look better, so fixed fb0e2c2fab6bbbd340ad78f088f7c1f3e30ea7c6